### ISSUE TITLE
No system tray icon on Wayland session of KDE Plasma desktop environment

### ISSUE THREAD

**Issue by glaubersm** [ugetdm/uget#45]
Hi
I use uGet 2.2.3 on Arch Linux running Wayland session of KDE Plasma 5.24.5 desktop environment.
uGget icon is missing from system tray despite "Always show tray icon' option is checked in 'User interface' section of uGet settings.

**Comment by userofryzen** [comment]
I have this issue too. I went to the uget website but it's with the cert expired.. so I didn't trust it and come here again to github. 
It's a coincidence that this issue is created with less than a day from another user! jajaja
I have noticed from a long time ago to be said too, Idk if did it ever work because I don't remember.. But until today, for the long time, I have not proposed to me to fill a bug

okey, a little bit different I'm using manjaro with latest updates and Plasma environment too(wayland session, I didn't test on xorg, but before it was working correct)

**Comment by userofryzen** [comment]
@glaubersm close this issue since it's a problem from arch(and manjaro) for reverting an upstream patch/decission for using X11 backend of GDK in wayland

more info: https://bugs.archlinux.org/task/70837

https://bugs.archlinux.org/task/74830?project=5&opened=20898 vote that bug pls so we can revert that decission.

**Comment by glaubersm** [comment]
Does uGet dev set a variable to force uGet to run on Xwayland (X11) and the Arch's packager reverted this?
uGet needs to support StatusNotifierItem specification. It works well with Wayland and KDE Plasma.

https://www.freedesktop.org/wiki/Specifications/StatusNotifierItem/

Apparently uGet uses the Freedesktop System tray specification that is deprecated on Wayland.

**Comment by userofryzen** [comment]
> Does uGet dev set a variable to force uGet to run on Xwayland (X11) and the Arch's packager reverted this? uGet needs to support StatusNotifierItem specification. It works well with Wayland and KDE Plasma.
yeah. maybe someone can develop the thing you have said but it appear that the project is a little bit slow theses days(since there is much time ago with the cert expired)
> 
> https://www.freedesktop.org/wiki/Specifications/StatusNotifierItem/
> 
> Apparently uGet uses the Freedesktop System tray specification that is deprecated on Wayland.
its appear but at least it works..

**Comment by userofryzen** [comment]
@glaubersm comment something on the bug page pls. Apart, vote and watch I think that way the bug will get more attention until today nobody has watched...

**Comment by glaubersm** [comment]
it's not Arch's bug. I have systray icon on Wayland when forcing uGet to run on Xwayland:

$ GDK_BACKEND=x11 uget-gtk

uGget needs StatusNotifierItem support to have systray icon when running natively on Wayland.
And the bug in Arch's tracker was already closed.

**Comment by userofryzen** [comment]
@glaubersm the problem is that arch decided to "force" uget to run on wayland mode because on the old bug you said is closed.

I mean that bug https://bugs.archlinux.org/task/74830?project=5&opened=20898 that I filled to revert that decision, do you understand?
Arch usually wants to use upstream things and not to change anything, that decision made is choking with that, so pls, vote the bug or something.
